libhb/nvenc_common.c

```diff
@@ -73,12 +127,44 @@ int hb_nvenc_h264_available()
 
 int hb_nvenc_h265_available()
 {
-    return hb_check_nvenc_available();
+    if (is_nvenc_hevc_available != -1)
+    {
+        return is_nvenc_hevc_available;
+    }
+    
+    if (!hb_check_nvenc_available()){
+        is_nvenc_hevc_available = 0;
+        return is_nvenc_hevc_available;
+    }
+    
+    if (hb_nvenc_get_cuda_version() >= 6.0) {
+        is_nvenc_hevc_available = 1;
+    }else {
+        is_nvenc_hevc_available = 0;
+    }
+        
+    return is_nvenc_hevc_available;
 }
 
 int hb_nvenc_av1_available()
 {
-    return hb_check_nvenc_available();
+    if (is_nvenc_av1_available != -1)
+    {
+        return is_nvenc_av1_available;
+    }
+    
+    if (!hb_check_nvenc_available()){
+        is_nvenc_av1_available = 0;
+        return is_nvenc_av1_available;
+    }
+    
+    if (hb_nvenc_get_cuda_version() >= 9) {
```

Comment:
sorry, my memory failed me: Ada lovelace is 8.9 https://en.wikipedia.org/wiki/CUDA. Hopper server cards are 9.0

Reply:
I also wonder how save doubles are for this?
What if it gets internally rounded to something like 8.8999999...?

Reply:
Yeah, that is also what I thought. See comment above.

Reply:
Changed it to 8.9.  Also changed the parsing code to avoid doing it a stupid way. It appears to work correctly.

My C is a bit lackluster however I think now there shouldn't be a precision issue with it?

Reply:
My concern about the precision is about how doubles/IEEE floating point numbers work.
A 8.9 or 6.0 as double might be internally represented by an approximation that's ever so slightly smaller or larger than the actual number.
I'm not sure if there is a guarantee that it's always slightly larger or something like that. But doing any kind of equality-comparison to a floating point number is generally a bad idea for that reason.
If it ends up actually being 8.899999999999999999912345... your >= would fail, even though the card actually is 8.9.

tl;dr: I'd use integers for this kind of check.
Something like: multiply major by 100 or 1000, and add the minor. Then compare >=809 or >=8009.

Reply:
It appears you get the major and minor as integers from CUDA in the first place anyway? I agree with the suggestion, floating-point equality comparisons of fractional values works really well until it doesn't, easier to "unnecessarily" make the comparison risk-free now than to possibly have to debug it later (without access to the hardware/software config that triggers it).

Reply:
Swapped it to an int comparison.